api/provider/aws/common.go

```diff
@@ -7,8 +7,48 @@ import (
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
 	"github.com/quintilesims/layer0/api/tag"
+	"github.com/quintilesims/layer0/common/errors"
 )
 
+func lookupEntityEnvironmentID(store tag.Store, entityType, entityID string) (string, error) {
+	tags, err := store.SelectByTypeAndID(entityType, entityID)
+	if err != nil {
+		return "", err
+	}
+
+	if tag, ok := tags.WithKey("environment_id").First(); ok {
+		return tag.Value, nil
+	}
+
+	return "", errors.Newf(errors.EnvironmentDoesNotExist, "Could not resolve environment ID for %s '%s'", entityType, entityID)
+}
+
+func lookupTaskDefinitionFamily(store tag.Store, deployID string) (string, error) {
```

Comment:
Can we merge these? Also, since we are looking up in our tags database, should the verbiage be in "layer0" terms? 
 `func lookupDeployNameAndVersion(store tag.Store, deployID string) (string, string, error)`

Reply:
For sure.

✅